### ISSUE TITLE
Error handling of non rdf-xml files

### ISSUE THREAD

**Issue by LaurensRietveld** [rdfjs/rdfxml-streaming-parser.js#12]
The parser won't throw an error when invalid rdf-xml (or non-xml) is submitted.

Some files return 0 statements from the parser, such as this one:
```
a b c
```

Other files return statements with predominantly blank nodes, such as 
```
<a:a> <b:b> <c:c> .
```
which outputs (after serializing to ntriples)
```
_:b1 a <a>.
_:b2 a <c>.
_:b1 <b> _:b2
```

Uploading other xml-like files (eg html, graphml), wont throw an error as well, but return many statements with many blank-nodes.

**Comment by rubensworks** [comment]
By default this parser does not perform any strict XML checking. Strict-mode can be enabled by passing `{ strict: true }` into the constructor of the parser.

If you do that, the `a b c` will throw an error.

Your second example still passes in strict-mode though, so that is definitely an error, as the `a`, `b` and `c` prefixes are undefined.

**Comment by rubensworks** [comment]
Fixed in v1.2.2.

**Comment by LaurensRietveld** [comment]
@rubensworks Thanks for the quick response :+1: 

I did notice an issue related to the error handling that pops up now. The stream continues, despite errors. I.e. the error event is emitted multiple times, and the input stream would keep passing data until eol.

How about:

- Change the error handling from emitting an error event directly to simply throwing the errors 
- Then, in the `_transform` function, catch all these errors and pass them to the callback:
```
  public _transform(chunk: any, encoding: string, callback: TransformCallback) {
    try {
      this.saxStream.write(chunk, encoding, callback);
    } catch(e) {
      callback(e)
    }
  }
```
Not sure whether this would cover all of your use cases, but it solved mine

**Comment by rubensworks** [comment]
Indeed, you're right.

I like your suggestion for the fix, it also simplifies and cleans up the code a bit.

Released as v1.2.3.
